detectors/aws/xray/EcsDetector.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Detectors\Aws\Xray;
+
+use OpenTelemetry\Sdk\Resource\ResourceConstants;
+use OpenTelemetry\Sdk\Resource\ResourceInfo;
+use OpenTelemetry\Sdk\Trace\Attributes;
+
+/**
+ * The AwsEcsDetector can be used to detect if a process is running in AWS
+ * ECS and return a {@link Resource} populated with data about the ECS
+ * plugins of AWS ˜X-Ray. Returns an empty Resource if detection fails.
+ */
+class EcsDetector
+{
+    private const ECS_METADATA_KEY_V4 = 'ECS_CONTAINER_METADATA_URI_V4';
+    private const ECS_METADATA_KEY_V3 = 'ECS_CONTAINER_METADATA_URI';
+
+    private const CONTAINER_ID_LENGTH = 64;
+    private const EMPTY_STRING = '';
+
+    public function __construct(GetProcessData $processData)
+    {
+        $this->processData = $processData;
+    }
+    
+    /**
+     * If running on ECS, runs getContainerId(), getClusterName(), and returns true.
+     * If not running on ECS, returns false.
+     */
+    public function detect(): ResourceInfo
+    {
+        // Check if running on ECS by looking for below environment variables
+        if (!getenv(self::ECS_METADATA_KEY_V4) && !getenv(self::ECS_METADATA_KEY_V3)) {
+            // TODO: add 'Process is not running on ECS' when logs are added
+            return ResourceInfo::emptyResource();
+        }
+
+        $hostName = $this->processData->getHostname();
+        $containerId = self::getContainerId();
```

Comment:
```suggestion
        $containerId = $this->getContainerId();
```

This isn't a static method, so be sure to invoke it on the current object instance.